demos/common/cpp/models/src/detection_model_centernet.cpp

```diff
@@ -0,0 +1,307 @@
+    std::string imageInputName = inputInfo.begin()->first;
+    inputsNames.push_back(imageInputName);
+    netInputHeight = getTensorHeight(inputDesc);
+    netInputWidth = getTensorWidth(inputDesc);
+
+    // --------------------------- Prepare output blobs -----------------------------------------------------
+    slog::info << "Checking that the outputs are as the demo expects" << slog::endl;
+
+    InferenceEngine::OutputsDataMap outputInfo(cnnNetwork.getOutputsInfo());
+    if (outputInfo.size() != 3) {
+        throw std::logic_error("This demo expect networks that have 3 outputs blobs");
+    }
+
+    const InferenceEngine::TensorDesc& outputDesc = outputInfo.begin()->second->getTensorDesc();
+    for (auto& output : outputInfo) {
+        output.second->setPrecision(InferenceEngine::Precision::FP32);
+        output.second->setLayout(InferenceEngine::Layout::NCHW);
+        outputsNames.push_back(output.first);
+    }
+
+}
+
+cv::Point2f getDir(const cv::Point2f& srcPoint, float rotRadius) {
+    float sn = sinf(rotRadius);
+    float cs = cosf(rotRadius);
+
+    cv::Point2f srcResult(0.0f, 0.0f);
+    srcResult.x = srcPoint.x * cs - srcPoint.y * sn;
+    srcResult.y = srcPoint.x * sn + srcPoint.y * cs;
+
+    return srcResult;
+}
+
+cv::Point2f get3rdPoint(const cv::Point2f& a, const cv::Point2f& b) {
+    cv::Point2f direct = a - b;
+    return b + cv::Point2f(-direct.y, direct.x);
+}
+
+cv::Mat getAffineTransform(float centerX, float centerY, int scale, float rot, size_t outputWidth, size_t outputHeight, bool inv = false) {
+    int srcW = scale;
+    float rotRad =  static_cast<float>(CV_PI) * rot / 180.0f;
+    auto srcDir = getDir({ 0.0f, -0.5f * srcW }, rotRad);
+    cv::Point2f dstDir(0.0f,  -0.5f * outputWidth);
+    std::vector<cv::Point2f> src(3, { 0.0f, 0.0f });
+    std::vector<cv::Point2f> dst(3, { 0.0f, 0.0f });
+
+    src[0] = { centerX, centerY };
+    src[1] = srcDir + src[0];
+    src[2] = get3rdPoint(src[0], src[1]);
+
+    dst[0] = { outputWidth * 0.5f, outputHeight * 0.5f };
+    dst[1] = dst[0] + dstDir;
+    dst[2] = get3rdPoint(dst[0], dst[1]);
+
+    cv::Mat trans;
+    if (inv) {
+        trans = cv::getAffineTransform(dst, src);
+    }
+    else {
+        trans = cv::getAffineTransform(src, dst);
+    }
+
+    return trans;
+}
+
+std::shared_ptr<InternalModelData> ModelCenterNet::preprocess(const InputData& inputData, InferenceEngine::InferRequest::Ptr& request) {
+    auto& img = inputData.asRef<ImageInputData>().inputImage;
+
+    int imgWidth = img.cols;
+    int imgHeight = img.rows;
+    float centerX = imgWidth / 2.0f;
+    float centerY = imgHeight / 2.0f;
+    int scale = std::max(imgWidth, imgHeight);
+
+    auto transInput = getAffineTransform(centerX, centerY, scale, 0, netInputWidth, netInputHeight);
+    cv::Mat resizedImg;
+    cv::warpAffine(img, resizedImg, transInput, cv::Size(netInputWidth, netInputHeight), cv::INTER_LINEAR);
```

Comment:
It feels it's just resizing keeping aspect ratio. @jkamelin, can you remember, why python version has this?

Reply:
Though I'm still interested in answer, even if affine transform can be replaced with resize, I think it's out of this PR scope.

Reply:
As far as I remember, this transformation is also used for mapping output points to original image. For input image this is probably equal to resizing + padding, though the original repo uses affine transform.

Reply:
I believe this transformation for the output points is needed to invert changes introduced in preprocess(). As for the original repo, may be it was simpler for authors to run already working transformations used for training for inference with zero rotation instead of writing another one.
This means to me it does make sense to replace that tricky transformations with resize some day.
Thanks!